### PR TITLE
feat: add multi-sort support to auto-grid

### PR DESCRIPTION
**Pull request by sissbruecker**
Changes the auto-grid's internal sort state to support multi-sort. 

For now I did not enable multi-sort by default, as I found it quite bothersome that you first have to remove sort from one column (two clicks) if you just want to sort a different column (third click). Let's discuss this separately, maybe we can use the new multi-sort on shift click mode as a compromise.

Closes https://github.com/vaadin/hilla/issues/1509

### CODE REVIEW

File: packages/ts/react-crud/test/autogrid.spec.tsx [hunk 1]
```diff
@@ -141,6 +151,66 @@ describe('@hilla/react-crud', () => {
...
+            { property: 'lastName', direction: Direction.DESC, ignoreCase: false },
+          ],
+        });
+        expect(getSortOrder(grid)).to.eql([
+          { property: 'firstName', direction: Direction.ASC },
+          { property: 'lastName', direction: Direction.DESC },
+        ]);
+      });
+    });
+
```
Comment (taefi) [inline packages/ts/react-crud/test/autogrid.spec.tsx:213]: Could we have a test to make sure when triggering the sort for `lastName` for the third time, it is excluded from the list of sorter columns?

**Review by taefi — CHANGES_REQUESTED** [review]

File: packages/ts/react-crud/test/autogrid.spec.tsx (hunk 1 above)
Comment (sissbruecker) [reply packages/ts/react-crud/test/autogrid.spec.tsx:213]: The sort function actually just sets the direction programmatically, it doesn't use clicks:
https://github.com/vaadin/hilla/blob/07cf6f3bf17bae2b03579ee15e59dbda0eb28842/packages/ts/react-crud/test/grid-test-helpers.ts#L66-L72

Comment (sissbruecker) [reply packages/ts/react-crud/test/autogrid.spec.tsx:213]: But still, removing the direction is a code path that should be tested, good point. Will add a test.

Comment (sissbruecker) [reply packages/ts/react-crud/test/autogrid.spec.tsx:220]: Done

**Review by taefi — APPROVED** [review]

**Pull request merged**